src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountingExtractReportServiceImpl.java

```diff
@@ -0,0 +1,303 @@
+package edu.cornell.kfs.concur.batch.service.impl;
+
+import java.io.File;
+import java.util.Date;
+import java.util.List;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+
+import edu.cornell.kfs.sys.service.ReportWriterService;
+import edu.cornell.kfs.sys.service.impl.ReportWriterTextServiceImpl;
+import edu.cornell.kfs.concur.batch.report.ConcurBatchReportMissingObjectCodeItem;
+import edu.cornell.kfs.concur.batch.report.ConcurBatchReportSummaryItem;
+import edu.cornell.kfs.concur.batch.report.ConcurBatchReportLineValidationErrorItem;
+import edu.cornell.kfs.concur.batch.report.ConcurStandardAccountingExtractBatchReportData;
+import edu.cornell.kfs.concur.batch.service.ConcurStandardAccountingExtractReportService;
+import edu.cornell.kfs.paymentworks.batch.report.SupplierUploadSummary;
+import edu.cornell.kfs.paymentworks.batch.report.SupplierUploadSummaryLine;
+
+public class ConcurStandardAccountingExtractReportServiceImpl implements ConcurStandardAccountingExtractReportService {
+    private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(ConcurStandardAccountingExtractReportServiceImpl.class);
+
+    protected ReportWriterService reportWriterService;
+
+    protected String fileNamePrefixFirstPart;
+    protected String fileNamePrefixSecondPart;
+    protected String reportTitle;
+    protected String reportConcurFileNameLabel;
+    protected String reimbursementsInExpenseReportLabel;
+    protected String cashAdvancesRelatedToExpenseReportsLabel;
+    protected String expensesPaidOnCorporateCordLabel;
+    protected String transactionsByPassedLabel;
+    protected String pdpRecordsProcessedLabel;
+    protected String reportValidationErrorsSubTitle;
+    protected String reportMissingObjectCodesSubTitle;
+    
+    public File generateSummaryReport(ConcurStandardAccountingExtractBatchReportData reportData) {
+        initializeReportTitleAndFileName(reportData);
+        writeSummarySubReport(reportData);
+        writeValidationErrorSubReport(reportData);
+        writeMissingObjectCodesSubReport(reportData);
+        finalizeReport();
+        return getReportWriterService().getReportFile();
+    }
+    
+    protected void initializeReportTitleAndFileName(ConcurStandardAccountingExtractBatchReportData reportData) {
+        String concurFileName = reportData.getConcurFileName();
+        if (StringUtils.isEmpty(concurFileName)) {
+            concurFileName = "UNKNOWN_SAE_FILENAME";
```

Comment:
It might be a good idea to create constants for some of these literals (and elsewhere in this file where practical).